valhalla/thor/map_matcher.h

```diff
@@ -22,14 +23,31 @@ namespace thor {
 
 class MapMatcher {
 public:
-  static std::vector<PathInfo>
+  static std::deque<std::pair<std::vector<PathInfo>, std::vector<const meili::EdgeSegment*>>>
```

Comment:
We're repeating this complex type in a lot of places: `std::deque<std::pair<std::vector<PathInfo>, std::vector<const meili::EdgeSegment*>>>`. Replacing this with a type alias might make the code easier to follow.

Reply:
ill document then. i think that a type alias makes these things harder to read (maybe im not a good enough word-smith to pick a good name?). instead what i will do is document the return type and what it refers to